catalog/service/src/main/java/org/projectnessie/catalog/service/resources/IcebergV1ApiResource.java

```diff
@@ -467,6 +482,138 @@ public LoadTableResponse updateTable(
     return buildLoadTableResponse(storedMetadata, commit.getTargetBranch(), false);
   }
 
+  private static class TableToCommit {
+    final UpdateTableRequest tableChange;
+    final ContentKey key;
+    IcebergTable content;
+    TableMetadata base;
+    int newVersion;
+    TableMetadata updatedMetadata;
+    TableMetadata storedMetadata;
+
+    TableToCommit(UpdateTableRequest tableChange, ContentKey key) {
+      this.tableChange = tableChange;
+      this.key = key;
+    }
+
+    ContentKey key() {
+      return key;
+    }
+  }
+
+  @Override
+  public void commitTransaction(String prefix, CommitTransactionRequest commitTransactionRequest)
```

Comment:
It's a bit dirty, and there's some code duplication with `updateTable`. But maybe it's "good enough" for now.